### PR TITLE
feat(button): Adds borderless and outline properties

### PR DESCRIPTION
**Pull request by diegonvs**

### CODE REVIEW

**Comment by matuzalemsteles** [comment]
I'd rather take it easy with that and review Issue #397 before merging.

**Comment by jbalsas** [comment]
Hey @matuzalemsteles, I reviewed our conversation in #397 and the final outcome was:

>After checking with @victorvalle and the Lexicon team, Borderless is indeed a variation, not a new type of button. They will proceed to document it in a different way and we can proceed to remove it from the clay component implementation.
>
>If someone wants a borderless button, as @pat270 mentioned, she can use the outline modifiers accordingly, or it will be styled by default based on context (like in a toolbar).

I think that should put `borderless` on the same level as other variants we already support such as `block`, `monospaced` or `small`. 

What do you think?

**Comment by matuzalemsteles** [comment]
I really do not see a problem putting it in, but what I've been thinking about ending up adding a lot of API to our components (even though they're the simplest) is that they can get so bloated that any future change would make it very painful to discontinue many if necessary.

So what I'm thinking is: what sets our API limit we can add is what the Lexicon is predicting... does that make sense to you?

File: packages/clay-button/src/index.tsx [hunk 1]
```diff
@@ -26,34 +31,44 @@ interface IProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
...
 	<button
-		{...otherProps}
-		className={classNames(className, 'btn', `btn-${displayType}`, {
+		className={classNames(className, 'btn', {
 			'btn-block': block,
 			'btn-monospaced': monospaced,
+			'btn-outline-borderless': borderless,
 			'btn-sm': small,
+			[`btn-${displayType}`]: displayType && !outline,
+			[`btn-outline-${displayType}`]: displayType && outline,
```
Comment (matuzalemsteles) [inline packages/clay-button/src/index.tsx]: For `borderless` to have effect it needs `btn-outline-*`and should disable `btn-${displayType}`.
```suggestion
			[`btn-outline-${displayType}`]: displayType && (outline || borderless),
```

**Pull request merged**